### PR TITLE
add Spark structured streaming option skip-overwrites to docs

### PR DESCRIPTION
**Pull request by SreeramGarlapati**

### CODE REVIEW

File: docs/versioned/spark/spark-structured-streaming.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ val df = spark.readStream
 ```
 
 {{< hint warning >}}
-Iceberg only supports reading data from append snapshots. Overwrite snapshots cannot be processed and will cause an exception. Similarly, delete snapshots will cause an exception by default, but deletes may be ignored by setting `streaming-skip-delete-snapshots=true`.
+Iceberg only supports reading data from snapshots of type `Append` or `Replace`. Snapshots of type `Delete` & `Overwrite` cannot be processed and will cause an exception. The spark options `streaming-skip-delete-snapshots=true` & `streaming-skip-overwrite-snapshots=true` may be used while reading from the iceberg tables where, there are known cases of deletes and overwrites performed on the table that can be ignored.
```
Comment (kbendick) [inline docs/versioned/spark/spark-structured-streaming.md:47]: Nit: The wording is a little weird for me in the last sentence.

I'd maybe say "The spark options `streaming-skip-delete-snapshots` and `streaming-skip-overwrite-snapshots` may be set to true to skip snapshots of type `Delete` and `Overwrite`, respectively, in cases where it's acceptable to ignore the table changes from those snapshots."

I'll leave that up to you though.

Comment (kbendick) [reply docs/versioned/spark/spark-structured-streaming.md:47]: Also, maybe don't use `&` (though if use that elsewhere in the docs then by all means ignore my comment 🙂  )

**Review by kbendick — APPROVED** [review]
+1. Left some feedback, which I'll leave up to you.

Overall it LGTM, though the language is a little unclear to me in the last sentence. But I'll leave that up to you.

**Comment by SreeramGarlapati** [comment]
thank you @kbendick - incorporated your feedback.

**Review by kbendick — APPROVED** [review]
This looks great. Thanks @SreeramGarlapati!

File: docs/versioned/spark/spark-structured-streaming.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ val df = spark.readStream
 ```
 
 {{< hint warning >}}
-Iceberg only supports reading data from append snapshots. Overwrite snapshots cannot be processed and will cause an exception. Similarly, delete snapshots will cause an exception by default, but deletes may be ignored by setting `streaming-skip-delete-snapshots=true`.
+Iceberg only supports reading data from snapshots of type `Append` or `Replace`. The snapshots of type `Delete` and `Overwrite` cannot be processed and will cause an exception. The spark options `streaming-skip-delete-snapshots` and `streaming-skip-overwrite-snapshots` may be set to `true` to skip snapshots of type `Delete` and `Overwrite`, respectively, in cases where it's acceptable to ignore the table changes from those snapshots.
```
Comment (rdblue) [inline docs/versioned/spark/spark-structured-streaming.md:47]: I think this should have fewer changes. It looks like this is just adding the `streaming-skip-overwrite-snapshots` property and I think it should do that using the same wording that was originally there. There isn't a need to change the sentence about delete snapshots.

Also, if you want to mention replace snapshots, we'll have to do some additional work. We don't want to assume that the reader is familiar with Iceberg internals or the spec, which is why we replace out originally. If you want to mention them, I think we should refer to them as "data rewrites that do not change table data".

Comment (kbendick) [reply docs/versioned/spark/spark-structured-streaming.md:47]: That's a fair point about `Replace` snapshots. Even as somebody working on Iceberg, I wasn't able to point out what was really meant by that until looking into it a bit more. It's definitely an implementation detail.

Comment (SreeramGarlapati) [reply docs/versioned/spark/spark-structured-streaming.md]: addressed the comments @rdblue @kbendick 
pl. take another look.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @SreeramGarlapati!

**Pull request merged**